harness/determined/common/experimental/experiment.py

```diff
@@ -498,26 +498,28 @@ def list_checkpoints(
         if not sort_by:
             sort_by = checkpoint.CheckpointSortBy.SEARCHER_METRIC
 
-        def get_with_offset(offset: int) -> bindings.v1GetExperimentCheckpointsResponse:
-            return bindings.get_GetExperimentCheckpoints(
-                self._session,
-                id=self._id,
-                limit=max_results,
-                offset=offset,
-                orderBy=order_by._to_bindings() if order_by else None,
-                sortByAttr=(
-                    sort_by._to_bindings()
-                    if isinstance(sort_by, checkpoint.CheckpointSortBy)
-                    else None
-                ),
-                sortByMetric=sort_by if isinstance(sort_by, str) else None,
-                states=[bindings.checkpointv1State.COMPLETED],
-            )
+        with self._session as sess:
```

Comment:
I'm uncomfortable modifying a session as a side effect when `list_checkpoints` is called. What if the session stored in `self._session` already had a stored `_http_session`?

Reply:
With `list_checkpoints` coded in this way, is there any way to use the same HTTP connection for two successive calls?

```python
with api.login(<details>) as sess:
  persistent_det = Determined.from_session(sess)
  persistent_det.list_checkpoints()  # closes the PoolManager in persistent_det
```